### ISSUE TITLE
Sort by Custom expression in form view not persistent

### ISSUE THREAD

**Issue by albinepro** [qgis/QGIS#49277]
### What is the bug or the crash?

When defining the sort order of table records in Form view by custom expression different to Preview expression, the sorting expression is not applied on reopening the table. The same happens to custom order of forms in child records when moving to next parent feature.
The expression seemt to be active when opening the expression editor, but you need to reconfirm it every time to take effect.

### Steps to reproduce the issue

1. Open attribute table
2. Switch to form view
3. Choose a custom expression to sort records
![obrazek](https://user-images.githubusercontent.com/58026508/177929543-818c4ba0-71f5-44da-96da-ddc6bed28ce7.png)
4. Set an expression by a different field than the default one
5. Confirm - desired order
6. Close and reopen the table form view - custom sorting is not applied

### Versions

QGIS version | 3.22.6-Białowieża

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by albinepro** [comment]
@DelazJ No, it doesn't. Always revert to to Preview Expression (ascending).

**Comment by Isaacson33** [comment]
This seems like the same regression that's been around for months... https://github.com/qgis/QGIS/issues/53894.

I don't know if it's related, but the issue with widow sizing https://github.com/qgis/QGIS/issues/53795 seems to have come about at the same time.

Basically, Qgis is no longer remembering pertinent data about the window after it is closed.

**Comment by BoswachterMarc** [comment]
For your information; I am having the same issue here (in a child form) in QGis version 3.34.1-Prizren

![image](https://github.com/qgis/QGIS/assets/57868975/5bd9ab4d-5922-486d-8e20-7cfbdb359e9c)

![image](https://github.com/qgis/QGIS/assets/57868975/a61e7da9-8cc3-4d32-b83b-4eceeff85ad8)

**Comment by ghtmtt** [comment]
not sure if this is the same same issue. But also the `Sort by preview expression` is not taken into account, unless the user changes the preview field and goes back to the previous one. Here a small video with the `World Map` layer and the `fid` column:

https://github.com/user-attachments/assets/5da00066-7580-4601-a5e2-fc4858fde46e

tested with

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /></head><body>

QGIS version | 3.41.0-Master
-- | --
QGIS code revision | fedd040708c

</body></html>
